File: VB/16_Print/SinglePage.vb
Imports Spire.Pdf
Imports Spire.Pdf.Print

Namespace SinglePage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument object.
			Dim doc As New PdfDocument()

			' Load the PDF file from the specified path.
			doc.LoadFromFile("..\..\..\..\..\..\Data\SinglePage.pdf")

			' Set the printing orientation to portrait (not landscape).
			doc.PrintSettings.Landscape = False

			' Select a page range from page 9 to page 15 for printing.
			doc.PrintSettings.SelectPageRange(9, 15)

			' Set the single page layout with custom scaling mode and maintain aspect ratio.
			doc.PrintSettings.SelectSinglePageLayout(PdfSinglePageScalingMode.CustomScale, True, 100)

			' Set the paper margins for printing to 10 units on all sides.
			doc.PrintSettings.SetPaperMargins(10, 10, 10, 10)

			' Print the document.
			doc.Print()

			' Close the document after printing.
			doc.Close()
		End Sub
	End Class
End Namespace

File: CS/16_Print/SinglePage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Print;

namespace SinglePage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf document
            PdfDocument doc = new PdfDocument();

            //Load a file
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\SinglePage.pdf");

            //Indicate whether the page is printed in landscape or portrait orientation
            doc.PrintSettings.Landscape = false;

            //Set print page range
            doc.PrintSettings.SelectPageRange(9, 15);

            //Select one page to one paper layout
            doc.PrintSettings.SelectSinglePageLayout(PdfSinglePageScalingMode.CustomScale, true, 100);

            //Set paper margins,measured in hundredths of an inch
            doc.PrintSettings.SetPaperMargins(10, 10, 10, 10);

            //Print document
            doc.Print();

            //Close the document
            doc.Close();
        }
    }
}
